core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -775,6 +793,9 @@ public Entity addNode(Location loc, Map<?,?> extraFlags) {
         entity.sensors().set(CLUSTER_MEMBER, true);
         entity.sensors().set(CLUSTER, this);
 
+        int nextClusterMemberId = getAttribute(NEXT_CLUSTER_MEMBER_ID).get();
+        entity.sensors().set(CLUSTER_MEMBER_ID, nextClusterMemberId);
```

Comment:
I'd say this is a config value, could even be passed in `createFlags`, to be accessible in the `init` method, but no strong feelings.